i18n/cmd/commands/transifex/pull_transifex.go

```diff
@@ -33,99 +37,227 @@ var pullTransifexCommand = &cobra.Command{
 }
 
 func getLanguages() []string {
-	req, err := http.NewRequest(
-		"GET",
-		fmt.Sprintf(
-			"https://www.transifex.com/api/2/project/%s/resource/%s/stats/",
-			project, resource,
-		), nil)
+	url := mainEndpoint + fmt.Sprintf("projects/o:%s:p:%s/languages", organization, project)
 
+	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
 		fmt.Println(err.Error())
 		os.Exit(1)
 	}
 
-	req.SetBasicAuth("api", apiKey)
-
-	resp, err := http.DefaultClient.Do(req)
+	addHeaders(req)
 
+	res, err := http.DefaultClient.Do(req)
 	if err != nil {
 		fmt.Println(err.Error())
 		os.Exit(1)
 	}
 
-	defer resp.Body.Close()
-
-	b, err := ioutil.ReadAll(resp.Body)
+	defer res.Body.Close()
+	b, err := ioutil.ReadAll(res.Body)
 	if err != nil {
 		fmt.Println(err.Error())
 		os.Exit(1)
 	}
 
-	var jsonResp map[string]interface{}
-	if err := json.Unmarshal(b, &jsonResp); err != nil {
+	var jsonRes map[string]interface{}
+	if err := json.Unmarshal(b, &jsonRes); err != nil {
 		fmt.Println(err.Error())
 		os.Exit(1)
 	}
 
-	var langs []string
-	for key := range jsonResp {
-		langs = append(langs, key)
+	var languages []string
+	data := jsonRes["data"].([]interface{})
+	for _, object := range data {
+		languageCode := object.(map[string]interface{})["attributes"].(map[string]interface{})["code"].(string)
+		languages = append(languages, languageCode)
 	}
-
-	return langs
+	return languages
 }
 
-func pullCatalog(cmd *cobra.Command, args []string) {
-	languages := getLanguages()
-	fmt.Println("translations found:", languages)
+// startTranslationDownload notifies Transifex that we want to start downloading
+// the resources file for the specified languageCode.
+// Returns an id to monitor the download status.
+func startTranslationDownload(languageCode string) string {
+	url := mainEndpoint + "resource_translations_async_downloads"
 
-	folder := args[0]
+	jsonData := map[string]interface{}{
+		"data": map[string]interface{}{
+			"relationships": map[string]interface{}{
+				"language": map[string]interface{}{
+					"data": map[string]string{
+						"id":   fmt.Sprintf("l:%s", languageCode),
+						"type": "languages",
+					},
+				},
+				"resource": map[string]interface{}{
+					"data": map[string]string{
+						"id":   fmt.Sprintf("o:%s:p:%s:r:%s", organization, project, resource),
+						"type": "resources",
+					},
+				},
+			},
+			"type": "resource_translations_async_downloads",
+		},
+	}
```

Comment:
Why not a struct? (same question for all the other JSON marshall/unmarshall)

Reply:
No reason really, I just started writing it like this and kept going but I see how written this way is much messier than using `struct`s.
I'll rewrite it using structs.